src/qml/components/SelectContactsList.qml

```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+    states: [
+        State {
+            name: "show header"
+            PropertyChanges {
+                target: d
+                headerHeight: d.expandedHeaderHeight
+                headerOpacity: 1
+            }
+        },
+        State {
+            name: "hide header"
+            PropertyChanges {
+                target: d
+                headerHeight: d.defaultHeaderHeight
+                headerOpacity: 0
+            }
+        }
+    ]
+
+    transitions: [
+        Transition {
+            NumberAnimation {
+                target: d
+                properties: "headerHeight, headerOpacity"
+                easing.type: Easing.InExpo
+                duration: Theme.animationDuration
+            }
+        }
+    ]
+
+    PropertyAnimation {
+        id: flickListView
+        target: modelListView
+        property: "contentY"
+        to: -d.expandedHeaderHeight
+        duration: Theme.animationDuration
+        easing.type: Easing.InExpo
+    }
+
+    Component {
+        id: contactListHeader
+
+        Item {
+            width: parent.width
+            height: d.headerHeight
+            opacity: d.headerOpacity
+            enabled: modelListView.state === "show header"
+
+            ListDelegate {
+                id: contactListDelegate
+                anchors.centerIn: parent
+                width: parent.width
+                height: d.defaultChatHeight
+
+                Avatar {
+                    id: avatar
+                    nickname: modelListView.newContactText
+                    Layout.alignment: Qt.AlignVCenter
+                }
+
+                Column {
+                    Layout.fillWidth: true
+                    clip: true
+
+                    Text {
+                        color: Theme.primaryTextColor
+                        font.pointSize: UiHelper.fixFontSz(15)
+                        text: modelListView.newContactText
+                    }
+
+                    Text {
+                        color: Theme.secondaryTextColor
+                        font.pointSize: UiHelper.fixFontSz(12)
+                        text: qsTr("Click here to select contact")
+                        width: parent.width
+                        elide: Text.ElideRight
+                        textFormat: Text.RichText
+                    }
+                }
+
+                onClicked: {
+                    if (d.model.selection.multiSelect) {
+                        d.model.toggleById(modelListView.newContactText)
+                    }
+                    contactSelected(modelListView.newContactText)
+                }
+            }
+        }
+    }
+
+    Component {
+        id: contactListComponent
+
+        ListDelegate {
+            id: contactListDelegate
+            leftInset: 0
+            rightInset: 0
+            width: modelListView.width
+            height: d.defaultChatHeight
+
+            Item { // we need this item!
```

Comment:
Add a reason

Reply:
ListDelegate is based on Layout, it doesn't work properly with std row, so I am using Item as a wrapper. Without it vertical center fails.